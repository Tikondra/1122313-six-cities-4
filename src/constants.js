export const PLACES = [
  `Beautiful & luxurious apartment at great location`,
  `Wood and stone place`,
  `Canal View Prinsengracht`,
  `Nice, cozy, warm big bed apartment`,
];

export const Settings = {
  COUNT_PLACES: 512,
};
