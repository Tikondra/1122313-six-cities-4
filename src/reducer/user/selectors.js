import NameSpace from "../name-space.js";

export const getAuthorizationStatus = (state) => state[NameSpace.USER].authorizationStatus;

export const getUserLogin = (state) => state[NameSpace.USER].user;
