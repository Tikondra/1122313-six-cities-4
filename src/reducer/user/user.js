import {ActionCreator as ActionCreatorApp} from "../app-state/app-state";

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  user: null,
};

const ActionType = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
};

const ActionCreator = {
  requireAuthorization: (status, user = null) => {
    return {
      type: ActionType.REQUIRED_AUTHORIZATION,
      payload: {status, user},
    };
  },
};

const Operation = {
  checkAuth: () => (dispatch, getState, api) => {
    return api.get(`/login`)
      .then((data) => {
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH, data.data.email));
      })
      .catch((err) => {
        throw err;
      });
  },

  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.login,
      password: authData.password,
    })
      .then((data) => {
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH, data.data.email));
        dispatch(ActionCreatorApp.toHome());
      });
  },
};

const reducer = (state = initialState, action) => {
  if (action.type === ActionType.REQUIRED_AUTHORIZATION) {
    return Object.assign({}, state, {
      authorizationStatus: action.payload.status,
      user: action.payload.user,
    });
  }

  return state;
};

export {ActionCreator, ActionType, reducer, AuthorizationStatus, Operation};
