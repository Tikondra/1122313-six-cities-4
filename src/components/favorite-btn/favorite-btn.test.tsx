import * as React from "react";
import * as renderer from "react-test-renderer";
import {BrowserRouter} from 'react-router-dom';
import FavoriteBtn from "./favorite-btn";
import {Format, PlacesListClass} from "../../constants";
import {AuthorizationStatus} from "../../reducer/user/user";

it(`should render FavoriteBtn`, function () {
  const tree = renderer
    .create(
        <BrowserRouter>
          <FavoriteBtn
            size={Format.PLACE_IMG_SIZE.NORMAL}
            type={PlacesListClass.MAIN_CARD}
            isActive={false}
            status={AuthorizationStatus.AUTH}
            onClickByFavorite={jest.fn}
          />
        </BrowserRouter>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
