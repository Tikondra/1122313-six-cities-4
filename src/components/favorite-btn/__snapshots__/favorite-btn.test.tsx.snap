// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FavoriteBtn 1`] = `
<button
  className="cities__bookmark-button button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height={210}
    width={260}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;
